node/runtime/pangolin/src/pallets/evm_.rs

```diff
@@ -133,13 +138,17 @@ where
 		};
 
 		match address {
-			// Ethereum precompiles
+			// Ethereum precompiles:
 			a if a == addr(1) => Some(ECRecover::execute(input, target_gas, context, is_static)),
 			a if a == addr(2) => Some(Sha256::execute(input, target_gas, context, is_static)),
 			a if a == addr(3) => Some(Ripemd160::execute(input, target_gas, context, is_static)),
 			a if a == addr(4) => Some(Identity::execute(input, target_gas, context, is_static)),
+			a if a == addr(5) => Some(Modexp::execute(input, target_gas, context, is_static)),
+			a if a == addr(6) => Some(Bn128Add::execute(input, target_gas, context, is_static)),
+			a if a == addr(7) => Some(Bn128Mul::execute(input, target_gas, context, is_static)),
+			a if a == addr(8) => Some(Bn128Pairing::execute(input, target_gas, context, is_static)),
 			a if a == addr(9) => Some(Blake2F::execute(input, target_gas, context, is_static)),
```

Comment:
For address 9, is it from ethereum or upstream frontier?

Reply:
Upstream Frontier

Reply:
https://ethereum.stackexchange.com/questions/15479/list-of-pre-compiled-contracts

And also checked with moonbeam too.